plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -959,7 +960,13 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
                 kwargs.update(kwargs[k])
                 del kwargs[k]
 
-        # Rename numa_enabled to numa. According the API documentation
+        # Map bool to string, according to API documentation.
+        if kwargs['agent'] is True:
+          kwargs['agent'] = 'enabled=1'
+        elif kwargs['agent'] is False:
+          kwargs['agent'] = 'enabled=0'
```

Comment:
Due to your change `kwargs['agent']` is now a string, and not a boolean. That means that neither of these two conditions will ever be true. This is very likely not what you intended.

(Also this change looks like it shouldn't be part of this PR.)